avm/res/web/connection/main.bicep

```diff
@@ -2,46 +2,102 @@ metadata name = 'API Connections'
+    id: subscriptionResourceId('Microsoft.Web/locations/managedApis', '${resourceLocation}', 'servicebus')
+  }
+'''
+})
+param api object?
+
 @description('Optional. Dictionary of custom parameter values for specific connections.')
 param customParameterValues object?
 
-@description('Required. Display name connection. Example: \'blobconnection\' when using blobs. It can change depending on the resource.')
+@description('Required. Display name connection. Example: `blobconnection` when using blobs. It can change depending on the resource.')
 param displayName string
 
-@description('Optional. Location of the deployment.')
-param location string = resourceGroup().location
-
 @description('Optional. Dictionary of nonsecret parameter values.')
 #disable-next-line secure-secrets-in-params // Not a secret
 param nonSecretParameterValues object?
 
-@description('Optional. Connection strings or access keys for connection. Example: \'accountName\' and \'accessKey\' when using blobs.  It can change depending on the resource.')
+@description('Optional. Connection strings or access keys for connection. Example: `accountName` and `accessKey` when using blobs. It can change depending on the resource.')
 @secure()
+@metadata({
+  example: '''
+    {
+      connectionString: 'listKeys('/subscriptions/<subscriptionId>/resourceGroups/<resourceGroupName>/Microsoft.ServiceBus/namespaces/AuthorizationRules/<serviceBusName>/RootManagedSharedAccessKey', '2023-01-01').primaryConnectionString'
+    }
+    {
+      rootfolder: fileshareConnection.rootfolder
+      authType: fileshareConnection.authType
+      // to add an object, use the any() function
+      gateway: any({
+        name: fileshareConnection.odgw.name
+        id: resourceId(fileshareConnection.odgw.resourceGroup, 'Microsoft.Web/connectionGateways', fileshareConnection.odgw.name)
+        type: 'Microsoft.Web/connectionGateways'
+      })
+      username: username
+      password: password
+    }
+  '''
+})
 param parameterValues object?
 
+@description('Optional. Additional parameter Value Set used for authentication settings.')
+@metadata({
+  example: '''
+  // for a Service Bus connection
+  {
+    name: 'managedIdentityAuth'
+    values: {
+      namespaceEndpoint: {
+        value: 'sb://${dependency.outputs.serviceBusEndpoint}'
+      }
+    }
+  }
+'''
+})
+param parameterValueSet object?
+
 @description('Optional. Array of role assignments to create.')
 param roleAssignments roleAssignmentType
 
 @description('Optional. Status of the connection.')
-param statuses array?
+param statuses array[]?
 
 @description('Optional. The lock settings of the service.')
 param lock lockType
 
+@metadata({
+  example: '''
+  {
+      key1: 'value1'
+      key2: 'value2'
+  }
+  '''
+})
 @description('Optional. Tags of the resource.')
 param tags object?
 
 @description('Optional. Links to test the API connection.')
-param testLinks array?
+param testLinks array[]?
```

Comment:
an array of arrays?

Reply:
This was to make sure it is crystal clear that the parameter is an array. But you are right, could be too much 🙃

Reply:
If not, maybe `object[][` would be better. But array of arrays is presumably just giving the wrong impression as it's definitely not that 😄

Reply:
They are object[].

Reply:
Thanks @ReneHezser, did you regenerate the files & rerun the tests? If yes, we should be good to go :)